RecoTracker/TrackProducer/test/FakeCPEFiller.cc

```diff
@@ -0,0 +1,185 @@
+//
+// Original Author:  Elisabetta Manca
+//         Created:  Wed, 01 Aug 2018 12:02:51 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDFilter.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackingRecHit/interface/TrackingRecHit.h"
+#include "DataFormats/TrackingRecHit/interface/TrackingRecHitFwd.h"
+#include "DataFormats/SiStripDetId/interface/SiStripDetId.h"
+#include "TrackingTools/PatternTools/interface/Trajectory.h"
+#include "TrackingTools/TrackFitters/interface/TrajectoryStateCombiner.h"
+
+#include "Geometry/CommonDetUnit/interface/GlobalTrackingGeometry.h"
+#include "Geometry/Records/interface/GlobalTrackingGeometryRecord.h"
+#include "DataFormats/GeometryCommonDetAlgo/interface/MeasurementPoint.h"
+#include "Geometry/CommonTopologies/interface/StripTopology.h"
+#include "Geometry/CommonTopologies/interface/TkRadialStripTopology.h"
+
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHitContainer.h"
+#include "SimTracker/TrackerHitAssociation/interface/TrackerHitAssociator.h"
+
+#include "RecoTracker/TransientTrackingRecHit/interface/TkTransientTrackingRecHitBuilder.h"
+#include "TrackingTools/Records/interface/TransientRecHitRecord.h"
+#include "RecoLocalTracker/ClusterParameterEstimator/interface/FakeCPE.h"
+#include "RecoLocalTracker/ClusterParameterEstimator/interface/PixelFakeCPE.h"
+#include "RecoLocalTracker/ClusterParameterEstimator/interface/StripFakeCPE.h"
+
+
+//
+// class declaration
+//
+
+class FakeCPEFiller final : public edm::one::EDFilter<> {
```

Comment:
Since this does not filter, and does not produce anything in the event, could it be an `edm::one::EDAnalyzer`?

If the worry is what `convertToUnscheduled()` does in the configuration, AFAIK it treats EDAnalyzer as it does EDFilter (i.e. does not remove them from Paths).

Reply:
yes was a worry about scheduling.
In principle could filter if the trajectory misses some hits (the matching is not 100% efficient)

Reply:
This is now a real filter (reject event if not all rechits matched a simhit